File: app/src/main/java/com/example/ep2024/data/di/RepositoryModule.java
package com.example.ep2024.data.di;

import com.example.ep2024.data.repository.UserRepositoryImpl;
import com.example.ep2024.domain.repository.ServiceRepository;
import com.example.ep2024.data.repository.ServiceRepositoryImpl;
import com.example.ep2024.domain.repository.UserRepository;

import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Module
public class RepositoryModule {

    @Provides
    @Singleton
    ServiceRepository provideServiceRepository() {
        return new ServiceRepositoryImpl();
    }

    @Provides
    @Singleton
    UserRepository provideUserRepository() {
        return new UserRepositoryImpl();
    }

}

File: data/src/main/java/com/example/data/di/RepositoryModule.kt
package com.example.data.di

import android.content.Context
import com.example.data.network.api.AuthApi
import com.example.data.repository.AuthRepositoryImpl
import com.example.data.storage.UserStorageService
import com.example.domain.repository.AuthRepository
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object RepositoryModule {

    @Provides
    @Singleton
    fun provideAuthRepository(
        api: AuthApi,
        userStorage: UserStorageService,
        @ApplicationContext context: Context
    ): AuthRepository {
        return AuthRepositoryImpl(api, userStorage, context)
    }

}